enrollment-server-onboarding-provider-innovatrics/src/main/java/com/wultra/app/onboardingserver/provider/innovatrics/InnovatricsApiService.java

```diff
@@ -145,4 +158,247 @@ public void deleteSelfie(final String customerId, final OwnerId ownerId) throws
+        }
+    }
+
+    /**
+     * Delete customer.
+     * @param customerId id of the customer.
+     * @param ownerId owner identification.
+     * @throws RemoteCommunicationException in case of 4xx or 5xx response status code.
+     */
+    public void deleteCustomer(final String customerId, final OwnerId ownerId) throws RemoteCommunicationException {
+        final String apiPath = "/api/v1/customers/%s".formatted(customerId);
+
+        try {
+            logger.info("Deleting customer {}, {}", customerId, ownerId);
+            logger.debug("Calling {}", apiPath);
+            final ResponseEntity<Void> response = restClient.delete(apiPath, new ParameterizedTypeReference<>() {});
+            logger.info("Got {} for deleting customer, {}", response.getStatusCode(), ownerId);
+            logger.debug("{} response status code: {}", apiPath, response.getStatusCode());
+            logger.trace("{} response: {}", apiPath, response);
+        } catch (RestClientException e) {
+            throw new RemoteCommunicationException("REST API call failed when deleting customer, statusCode=%s, responseBody='%s'".formatted(e.getStatusCode(), e.getResponse()), e);
+        }
+    }
+
+    /**
+     * Delete customer's document.
+     * @param customerId id of the customer.
+     * @param ownerId owner identification.
+     * @throws RemoteCommunicationException in case of 4xx or 5xx response status code.
+     */
+    public void deleteDocument(final String customerId, final OwnerId ownerId) throws RemoteCommunicationException {
+        final String apiPath = "/api/v1/customers/%s/document".formatted(customerId);
+
+        try {
+            logger.info("Deleting document of customer {}, {}", customerId, ownerId);
+            logger.debug("Calling {}", apiPath);
+            final ResponseEntity<Void> response = restClient.delete(apiPath, new ParameterizedTypeReference<>() {});
+            logger.info("Got {} for deleting customer's document, {}", response.getStatusCode(), ownerId);
+            logger.debug("{} response status code: {}", apiPath, response.getStatusCode());
+            logger.trace("{} response: {}", apiPath, response);
+        } catch (RestClientException e) {
+            throw new RemoteCommunicationException("REST API call failed when deleting customer's document, statusCode=%s, responseBody='%s'".formatted(e.getStatusCode(), e.getResponse()), e);
+        }
+    }
+
+    /**
+     * Converts internal DocumentType enum to string value used by Innovatrics.
+     * @param type represents type of document.
+     * @return document type as a string value.
+     */
+    private static String convertType(DocumentType type) {
+        return switch (type) {
+            case ID_CARD -> "identity-card";
+            case PASSPORT -> "passport";
+            case DRIVING_LICENSE -> "drivers-licence";
+            default -> throw new IllegalStateException("Unsupported documentType " + type);
```

Comment:
An `IllegalStateException` will lead to a 500 Internal server error. Can we handle this case with an exception that is handled in an exception handler?